app/routes.py

```diff
@@ -1,2 +1,280 @@
+    data={'channel': f"{os.environ.get('SLACK_CHANNEL')}", 'text': text})
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def handle_tasks():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(title= request_body["title"],
+    description= request_body["description"],
+    completed_at= request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": {
+        "id": new_task.task_id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": is_complete_function(new_task.completed_at)}}), 201
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def tasks_index():
+    
+    sort_query = request.args.get("sort")
+    tasks_response = []
+    if sort_query == "asc":
+        tasks = Task.query.order_by(asc(Task.title))
+
+        for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+        return jsonify(tasks_response), 200
+
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(desc(Task.title))
+
+        for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+        return jsonify(tasks_response), 200
+
+    else:
+        tasks = Task.query.all()
+    
+        if tasks is None:
+            return jsonify(tasks_response), 200
+    
+        else:
+            for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+            return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def handle_single_task(task_id):
+
+    task = Task.query.get_or_404(task_id)
+    
+    if task.goal_id:
+        return jsonify({"task": {
+        "id": task.task_id,
+        "goal_id": task.goal_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": is_complete_function(task.completed_at)}}), 200
+
+    return jsonify({"task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": is_complete_function(task.completed_at)}}), 200
```

Comment:
we can shorten this up too since we are repeating a few lines:

```suggestion
    task_response = {
        "id": task.task_id,
        "title": task.title,
        "description": task.description,
        "is_complete": is_complete_function(task.completed_at)
    }
    if task.goal_id:
        task_response["goal_id"] = task.goal_id

    return jsonify({"task":  task_response}), 200
```